erpnext/selling/doctype/customer/customer.py

```diff
@@ -238,6 +239,64 @@ def create_contact(contact, party_type, party, email):
 	contact.append('links', dict(link_doctype=party_type, link_name=party))
 	contact.insert()
 
+@frappe.whitelist()
+def make_quotation(source_name, target_doc=None):
+
+	def set_missing_values(source, target):
+		_set_missing_values(source,target)
+
+	target_doc = get_mapped_doc("Customer", source_name, 
+		{"Customer": {
+			"doctype": "Quotation",
+			"field_map": {
+				"name":"party_name"
+			}
+		}}, target_doc, set_missing_values)
+
+	target_doc.quotation_to = "Customer"
+	target_doc.run_method("set_missing_values")
+	target_doc.run_method("set_other_charges")
+	target_doc.run_method("calculate_taxes_and_totals")
+
+	target_doc.selling_price_list = frappe.get_doc("Customer",source_name).default_price_list
```

Comment:
Why get the entire doc just to get the price list use `get_value` instead. Also override the default selling price list only if default price list for customer is available otherwise let it fetch the default price list.